### ISSUE TITLE
createNode in different plugin with same node type overwritted?

### ISSUE THREAD

**Issue by kerorojason** [gatsbyjs/gatsby#22935]
### Description
Hi everyone, I'm new in Gatsby.js and I'm trying to build my own blog with it. In addition to supporting markdown format, I also want to sync my Notion page into my app so that I can use Notion as my CMS and automatically update to my blog. And I found a plugin called [gatsby-source-notion-database](https://github.com/mayneyao/gatsby-source-notion-database) which helps me do that by first crawl the Notion page and put the html directly in my app.
However, I faced a problem when I'm trying to list all markdown pages and notion pages all together on my page-list page. I'm using [this Gatsby starter](https://github.com/alxshelepenok/gatsby-starter-lumen) and when it's creating page-list and some paginating stuff, it will query with `allMarkdownRemark`. Therefore I changed the type of the created node in gatsby-source-notion-database plugin to markdownRemark, as shown below:

```
const nodeMeta = {
    id: createNodeId(itemData[key]),
    parent: null,
    children: [],
    internal: {
        type:'markdownRemark',
        mediaType: `text/html`,
        content: nodeContent,
        contentDigest: createContentDigest(data)
    },
}
const node = Object.assign({}, data, nodeMeta)
createNode(node)
```

I tried to create node with type 'markdownRemark', so that I can query them easily to get all my pages. However, by doing so, it overwrites the markdownRemark nodes I previously create in `gatsby-transformer-remark` plugin.

I've been stuck in this problem for two days. Is there any way to fix it? Thx!

### Expected result
When I run the query below, it should shows both markdown pages and notion pages.
```
query MyQuery {
  allMarkdownRemark {
    edges {
      node {
        id
      }
    }
  }
}
```

### Actual result
I can only get notion pages.

### My environment
```
System:
    OS: macOS Mojave 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 13.8.0 - /usr/local/bin/node
    npm: 6.13.7 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  npmPackages:
    gatsby: ^2.20.8 => 2.20.10
    gatsby-link: ^2.3.1 => 2.3.1
    gatsby-plugin-catch-links: ^2.2.1 => 2.2.1
    gatsby-plugin-feed: ^2.4.1 => 2.4.1
    gatsby-plugin-flow: ^1.2.1 => 1.2.1
    gatsby-plugin-google-gtag: ^1.2.1 => 1.2.1
    gatsby-plugin-manifest: ^2.3.3 => 2.3.3
    gatsby-plugin-netlify: ^2.2.1 => 2.2.1
    gatsby-plugin-netlify-cms: ^4.2.1 => 4.2.2
    gatsby-plugin-offline: ^3.1.2 => 3.1.2
    gatsby-plugin-optimize-svgs: ^1.0.4 => 1.0.4
    gatsby-plugin-react-helmet: ^3.2.1 => 3.2.1
    gatsby-plugin-sass: ^2.2.1 => 2.2.1
    gatsby-plugin-sharp: ^2.5.3 => 2.5.3
    gatsby-plugin-sitemap: ^2.3.1 => 2.3.1
    gatsby-remark-autolink-headers: ^2.2.1 => 2.2.1
    gatsby-remark-copy-linked-files: ^2.2.1 => 2.2.1
    gatsby-remark-external-links: 0.0.4 => 0.0.4
    gatsby-remark-images: ^3.2.1 => 3.2.1
    gatsby-remark-katex: ^3.2.1 => 3.2.1
    gatsby-remark-prismjs: ^3.4.1 => 3.4.1
    gatsby-remark-relative-images: ^0.3.0 => 0.3.0
    gatsby-remark-responsive-iframe: ^2.3.1 => 2.3.1
    gatsby-remark-smartypants: ^2.2.1 => 2.2.1
    gatsby-source-filesystem: ^2.2.2 => 2.2.2
    gatsby-source-notion-database: ^0.6.9 => 0.6.9
    gatsby-transformer-remark: ^2.7.1 => 2.7.1
    gatsby-transformer-sharp: ^2.4.3 => 2.4.3
  npmGlobalPackages:
    gatsby: 2.20.10
```

**Comment by LekoArts** [comment]
Thank you for opening this!

TL;DR is: You shouldn't do what you tried to do. As you've seen overriding other types won't work and creates problems. You should let the Notion source plugin creates its nodes under its own type, query it together on your page, e.g.

```graphql
query MyQuery {
  allMarkdownRemark {
    nodes {}
  }
  allPosts {
    nodes {}
  }
}
```

And then use JS to merge those two arrays into one array. The merging/filtering you'll do in the head of your React component will still happen at build time.

We're marking this issue as answered and closing it for now but please feel free to comment here if you would like to continue this discussion. We also recommend heading over to our [communities](https://www.gatsbyjs.org/contributing/community/#where-to-get-support) if you have questions that are not bug reports or feature requests. We hope we managed to help and thank you for using Gatsby!
